### PR TITLE
Fixed Issues with Daily Island Missions

### PR DESCRIPTION
**Pull request by PeachesMLG**
Fixes https://github.com/Iridium-Development/IridiumSkyblock/issues/499

### CODE REVIEW

File: src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java [hunk 1]
```diff
@@ -808,6 +809,28 @@ public synchronized IslandMission getIslandMission(@NotNull Island island, @NotN
         }
     }
 
+    private String getDailyIslandMissions(@NotNull Island island, int index) {
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java:812]: It's just one mission, isn't it? Name it `getDailyIslandMission` then

File: src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java [hunk 2]
```diff
@@ -808,6 +809,28 @@ public synchronized IslandMission getIslandMission(@NotNull Island island, @NotN
...
+        List<String> islandMissions = IridiumSkyblock.getInstance().getDatabaseManager().getIslandMissionTableManager().getEntries(island).stream()
+                .filter(islandMission -> islandMission.getType() == Mission.MissionType.DAILY)
+                .map(IslandMission::getMissionName)
+                .distinct()
+                .collect(Collectors.toList());
+        if (islandMissions.size() > index) {
+            return islandMissions.get(index);
+        }
+        Random random = new Random();
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java:821]: Use ThreadLocalRandom (via `ThreadLocalRandom.getCurrent()`)

File: src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java [hunk 3]
```diff
@@ -808,6 +809,28 @@ public synchronized IslandMission getIslandMission(@NotNull Island island, @NotN
...
 
+    private String getDailyIslandMissions(@NotNull Island island, int index) {
+        List<String> islandMissions = IridiumSkyblock.getInstance().getDatabaseManager().getIslandMissionTableManager().getEntries(island).stream()
+                .filter(islandMission -> islandMission.getType() == Mission.MissionType.DAILY)
+                .map(IslandMission::getMissionName)
+                .distinct()
+                .collect(Collectors.toList());
+        if (islandMissions.size() > index) {
+            return islandMissions.get(index);
+        }
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java:820]: We need more empty lines
```suggestion

        if (islandMissions.size() > index) {
            return islandMissions.get(index);
        }

```

File: src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java [hunk 4]
```diff
@@ -808,6 +809,28 @@ public synchronized IslandMission getIslandMission(@NotNull Island island, @NotN
...
+        Random random = new Random();
+        List<String> missionList = IridiumSkyblock.getInstance().getMissionsList().keySet().stream()
+                .filter(mission -> IridiumSkyblock.getInstance().getMissionsList().get(mission).getMissionType() == Mission.MissionType.DAILY)
+                .filter(mission -> islandMissions.stream().noneMatch(m -> m.equals(mission)))
+                .collect(Collectors.toList());
+        String key = missionList.get(random.nextInt(missionList.size()));
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java:826]: Even more
```suggestion

        String key = missionList.get(random.nextInt(missionList.size()));
```

File: src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java [hunk 5]
```diff
@@ -808,6 +809,28 @@ public synchronized IslandMission getIslandMission(@NotNull Island island, @NotN
...
+        Mission mission = IridiumSkyblock.getInstance().getMissionsList().get(key);
+        for (int i = 0; i < mission.getMissions().size(); i++) {
+            IridiumSkyblock.getInstance().getDatabaseManager().getIslandMissionTableManager().addEntry(new IslandMission(island, mission, key, i));
+        }
+        return key;
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java:831]: And more
```suggestion

        return key;
```

**Review by dlsf — CHANGES_REQUESTED** [review]

File: src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java [hunk 6]
```diff
@@ -808,6 +810,31 @@ public synchronized IslandMission getIslandMission(@NotNull Island island, @NotN
...
+
+        if (islandMissions.size() > index) {
+            return islandMissions.get(index);
+        }
+
+        ThreadLocalRandom random = ThreadLocalRandom.current();
+        List<String> availableMissions = IridiumSkyblock.getInstance().getMissionsList().keySet().stream()
+                .filter(mission -> IridiumSkyblock.getInstance().getMissionsList().get(mission).getMissionType() == Mission.MissionType.DAILY)
+                .filter(mission -> islandMissions.stream().noneMatch(m -> m.equals(mission)))
+                .collect(Collectors.toList());
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java:828]: Forgot an empty line
```suggestion
                .collect(Collectors.toList());
                
```

**Review by dlsf — APPROVED** [review]

File: src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java (hunk 2 above)
Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java]: Why das? (Just for asking)

**Pull request merged**